### PR TITLE
Use filter_entity and filter_attribute context

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Use `filter_entity` and `filter_attribute` context in automation triggers and condition and weather card.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-selector/ha-selector-attribute.ts [hunk 1]
```diff
@@ -23,15 +23,21 @@ export class HaSelectorAttribute extends SubscribeMixin(LitElement) {
...
-    filter_entity?: string;
+    filter_entity?: string | string[];
   };
 
+  get _filterEntity(): string | undefined {
+    if (!this.context?.filter_entity) return undefined;
+    return Array.isArray(this.context.filter_entity)
+      ? this.context.filter_entity[0]
+      : this.context.filter_entity;
+  }
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-attribute.ts:34]: I don't really like this... should we even allow multiple in the UI? It is confusing with things like this where we only look at the first entity...

Comment (piitaya) [reply src/components/ha-selector/ha-selector-attribute.ts:34]: That's what we had before in automations trigger/condition : 
```ts
entity_id: entityId ? entityId[0] : undefined,
```

Comment (piitaya) [reply src/components/ha-selector/ha-selector-attribute.ts:34]: UI supports multiple entities.

CleanShot 2023-02-08 at 15 28 05

Comment (piitaya) [reply src/components/ha-selector/ha-selector-attribute.ts]: I removed the usage of filter entity/attributes for multiple entity_id

**Review by bramkragten — APPROVED** [review]

**Pull request merged**